app/assets/javascripts/backbone/views/calendar_step_modal_view.coffee

```diff
@@ -1,22 +1,32 @@
 class MS.Views.CalendarStepModalView extends Backbone.View
 
-  el: '.calendar'
+  el: '#calendar-step-modal'
+
+  initialize: (params) ->
+    @context = params.context
+    if params.id
+      @id = params.id
+      @showStep()
+    else
+      @loanId = params.loanId
+      @showNewStep()
 
   events:
-    'click .loan-calendar .cal-step': 'showStepModal'
-    'click a.action-delete': 'deleteStep'
+    'click a.action-delete': 'hideModal'
+    'click #new_project_step .cancel': 'hideModal'
+
+  showStep: ->
+    $.get '/admin/project_steps/' + @id, context: @context, (html) =>
```

Comment:
Please use string interpolation in cases like these: `"/admin/project_steps/#{@id}"`